pkg/lib/aws/cloudwatch.go

```diff
@@ -60,7 +64,291 @@ func (c *Client) TagLogGroup(logGroup string, tagMap map[string]string) error {
 
 	if err != nil {
 		return errors.Wrap(err, "failed to add tags to log group", logGroup)
+
+	}
+
+	return nil
+}
+
+// UpdateDashboard updates existing dashboard by adding API name title, In flight request time, latency and request status metric
+func (c *Client) UpdateDashboard(dashboardName string, dashboardRegion string, nameAPI string, apiID string) error {
+
+	// get current dashboard form cloudwatch
+	currDashboardOutput, err := c.CloudWatch().GetDashboard(&cloudwatch.GetDashboardInput{
+		DashboardName: aws.String(dashboardName),
+	})
+	if err != nil {
+		return errors.Wrap(err)
+	}
+	// get body string from GetDashboard return object
+	currDashboardString := *currDashboardOutput.DashboardBody
+	// get lowest element of cloudwatch. Needed to place metrics below all existing metrics
+	highestY, err := getHighestYDashboard(currDashboardString)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+
+	// create widgets for title and metrics
+	apiTitleWidget := createTextWidget(1, highestY+1, 22, 1, "## API: "+nameAPI)
+	inFlightWidget := createMetricWidget(1, highestY+2, 11, 6, inFlightMetric(dashboardName, nameAPI), "In flight requests", "Sum", dashboardRegion)
+	latencyWidgetp99 := createMetricWidget(12, highestY+2, 11, 6, latencyMetric(dashboardName, nameAPI, apiID), "p99 of request response time (60s)", "p99", dashboardRegion)
+	latencyWidgetp50 := createMetricWidget(12, highestY+8, 11, 6, latencyMetric(dashboardName, nameAPI, apiID), "median request response time (60s)", "p50", dashboardRegion)
+	statCodeWidget := createMetricWidget(1, highestY+8, 11, 6, statCodeMetric(dashboardName, nameAPI, apiID), "Status Code", "Sum", dashboardRegion)
+
+	//define interface to unmarshal received body string
+	var currDashboardHolder interface{}
```

Comment:
I thought about that too. The issue is that we can create a struct only for the general cloudwatch structure. This would help with this casting `currDashboard := currDashboardHolder.(map[string]interface{})`. But for this casting ` currDashboard["widgets"].([]interface{})` we would have to create a struct for each widget type and this I think would increase the logic.

Reply:
Based on a quick scan of their documentation, it looks like the widgets have the same schema, except for `properties`. Would it be possible to leave the `properties` field as an `interface{}`, and would that reduce the parsing complexity be enough to be worth it? e.g. something like this:

```go
type Widget struct {
	Type       string      `json:"type"`
	X          int         `json:"x"`
	Y          int         `json:"y"`
	Width      int         `json:"width"`
	Height     int         `json:"height"`
	Properties interface{} `json:"properties"`
}
```

Reply:
oh yes. correct. already late...

thanks for the help

Reply:
haha no worries, thank you!

Reply:
This was a very good idea. Code much nicer now.